src/GADM.jl

```diff
@@ -52,86 +53,65 @@ function dataread(path)
     !isnothing(data) ? data : throw(ArgumentError("failed to read data from disk"))
 end
 
+"""
+    getdataset(country)
+
+Downloads and extracts dataset of the given country code
+"""
+function getdataset(country)
+    isvalidcode(country) || throw(ArgumentError("please provide standard ISO 3 country codes"))
+    data = country |> download |> dataread
+end
+
 """
     getlayer(data, level) 
 
 Get layer of the desired `level` from the `data`.
 """
 function getlayer(data, level)
     nlayers = ArchGDAL.nlayer(data)
-    for l=0:nlayers-1
+    for l = 0:nlayers - 1
         layer = ArchGDAL.getlayer(data, l)
         lname = ArchGDAL.getname(layer)
         llevel = last(split(lname, "_"))
         string(level) == llevel && return layer
     end
-    throw(ArgumentError("valid levels are 0...$(nlayers-1)"))
+    throw(ArgumentError("asked for level $(level), valid levels are 0-$(nlayers - 1)"))
 end
 
-"""
-    get(country, subregions...)
-
-Returns the MULTIPOLYGON data for the requested region.
-
-Input: ISO 3 country code, and further subdivisions.
-
-## Examples  
-  
-```julia
-get("IND")
-get("IND", "Uttar Pradesh")
-get("IND", "Uttar Pradesh", "Lucknow")
-```
-"""
-function get(country, subregions...) 
-    isvalidcode(country) || throw(ArgumentError("please provide standard ISO 3 country codes"))
-
-    data = country |> download |> dataread
+function get(country, subregions...;children=false)
+    data = getdataset(country)
+    nlayers = ArchGDAL.nlayer(data)
 
-    function multipolygon(geom)
-       if GeoInterface.geotype(geom) == :Polygon
-        mp = ArchGDAL.createmultipolygon()
-        ArchGDAL.addgeom!(mp, geom)
-        return mp
-       else
-        return geom
-       end
+    function filterlayer(layer, key, value, all=false)
+        table = ArchGDAL.Table(layer)
+        filtered_rows = []
+        for row in Tables.rows(table)
+            field = row[Symbol(key)]
+            if all || occursin(lowercase(value), lowercase(field))
+                push!(filtered_rows, row)
+            end
+        end
+        return Tables.columntable(filtered_rows)
     end
 
-    level = length(subregions)
+    parent_level = length(subregions)
+    parent_level >= nlayers && throw(ArgumentError("more subregions provided than actual")) 
+    parent_name = isempty(subregions) ? "" : last(subregions)
+    parent_layer = getlayer(data, parent_level)
+    parent = filterlayer(parent_layer, "NAME_$(parent_level)", parent_name, iszero(parent_level))
+    isempty(parent) && throw(ArgumentError("could not find required region"))
```

Comment:
A side comment: I feel this code is so verbose with underscores everywhere. Try to put some time to figure out a short variable name that is clean and to the point. It can simplify the code and facilitate the life of the potential contributor.